__tests__/app.test.js

```diff
@@ -63,20 +63,124 @@ describe("GET/api/articles", () => {
+              comment_count: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+  test("200: should accept a sort_by query which sorts the article by any valid column (defaults to descending)", () => {
+    return request(app)
+      .get("/api/articles?sort_by=author")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("author", { descending: true });
+      });
+  });
+  test("200: articles should be sorted by date if no query is provided (defaults to descending) ", () => {
     return request(app)
       .get("/api/articles")
       .expect(200)
       .then(({ body: { articles } }) => {
         expect(articles).toBeSortedBy("created_at", { descending: true });
       });
   });
-  test("404: non-existent route", () => {
+  test("200: should accept an order query set to 'asc', which sorts the article in an ascending order", () => {
     return request(app)
-      .get("/api/articl")
-      .expect(404)
-      .then(({ body }) => {
-        expect(body).toEqual({ msg: "Path Not Found" });
+      .get("/api/articles?sort_by=author&order=asc")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("author");
+      });
+  });
+  test("200: should accept an order query set to 'desc' by default, which sorts the article in a descending order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=author")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("author", { descending: true });
+      });
+  });
+  test("400: invalid value requested by the client for order query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=author&order=ascending")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: non-existent column requested by the client for sort_by query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=banana")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("200: responds with an array of articles given all three valid queries", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch&sort_by=author&order=asc")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles).toBeSortedBy("author");
+        articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              topic: "mitch",
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              comment_count: expect.any(String),
+            })
+          );
+        });
```

Comment:
Please add another test where the topic doesn't exist, e. g. `?topic=bananas`. Expecting a status 404, 'not found'.

Reply:
I have already tested for this situation which returns an empty array and a status 200 if the topic does not exist. 
I was thinking of creating a variable validTopicQueries = ['name of all the topics inside this array'], and then having an if statement where if the values inside validTopicQueries does not include in the topic, promise reject it with a 404.
But I had a conflicting thought about it where the topics in out testData and in our devData has different topics. So the test would not pass in different environments. But am I wrong to think in that way? And just carry on with the above solution for a 404 error?